### DISCUSSION TITLE
Linq2Acad for AutoCAD 2025

### DISCUSSION THREAD

**Discussion by kdub-nz** [wtertinek/Linq2Acad#47] [General]
Hi Wolfgang,
Any progress on a build for Linq2Acad for AutoCAD 2025 ?

Regards,

**Comment by wtertinek** [comment]
Hi Kerry, unfortunately I have absolutely no time for Linq2Acad at the moment. I hope that will change in a few weeks.

**Comment by joseguiaCES** [comment]
I forked up and added a "stripped" down version of the Linq2Acad. This was definitely a learning process for me, it's the first time I've posted something back I toyed with so ... your mileage may vary, the code works great in the projects I'm using it on. 
Thanks Wolfgang - I learned a lot from your style and use.
https://github.com/joseguiaCES/Linq2Acad25

actions:
removed samples/tests projects
removed debug/release for prior versions
added acad 25 nuget 
disabled/enabled warning in Helpers.cs for #pragma warning disable SYSLIB0011
upgraded the project using the "visual studio -> upgrade" option
updated nuget packages

**Reply by wtertinek** [reply]
Hi joseguiaCES, thanks for the code you shared with us. As far as I can see, the csproj file is the main thing that changed (besides some minor other changes). So I will take your csproj file as a template for integrating it with the current csproj file.

**Reply by joseguiaCES** [reply]
Excellent! - I tend to follow the migration instructions shared with others in the forums so I wasn't sure what all changes the "upgrade assistant" would actually apply. But you would be correct, outside of running that, there were actually very few and minor changes overall. I am going to manually run all of the sample C# exercises that you created, 17 of them I believe, to make sure everything is in fact intact. I ported the samples into a test class in my code to re-use snippets. Will report back shortly.

**Reply by joseguiaCES** [reply]
I'd say it was a success, I use NLOG as my logger of choice :
 (feel the need to say, obvious error checking should be added for the unaware)

9:29.TstCmd.WrtMsg-This sample removes all entities from the model space 
9:29.TstCmd.WrtMsg-Model space cleared 
9:30.TstCmd.WrtMsg-This sample removes all BlockReferences from the model space 
9:30.TstCmd.WrtMsg-All block references removed from model space 
9:31.TstCmd.WrtMsg-This sample adds a line to the model space 
9:31.TstCmd.WrtMsg-Line added to model space 
9:31.TstCmd.WrtMsg-This sample creates a new layer 
9:31.TstCmd.WrtMsg-Layer TestLINQ2A created 
9:32.TstCmd.WrtMsg-This sample prints all layer names 
9:32.TstCmd.WrtMsg-0 
9:32.TstCmd.WrtMsg-OUTLINE 
9:32.TstCmd.WrtMsg-DIM 
9:32.TstCmd.WrtMsg-BLOCKS 
9:32.TstCmd.WrtMsg-Defpoints 
9:32.TstCmd.WrtMsg-QC 
9:32.TstCmd.WrtMsg-BLOCK 
9:32.TstCmd.WrtMsg-TestLINQ2A 
9:32.TstCmd.WrtMsg-This sample turns off all layers, except the one the user enters 
9:32.TstCmd.WrtMsg-All layers turned off, except door 
9:33.TstCmd.WrtMsg-This sample creates a layer and adds all red lines in the model space to it 
9:33.TstCmd.WrtMsg-This sample creates a layer and adds all red lines in the model space to it 
9:33.TstCmd.WrtMsg-All red lines moved to new layer Dim2 
9:34.TstCmd.WrtMsg-This sample moves entities from one layer to another 
9:34.TstCmd.WrtMsg-This sample moves entities from one layer to another 
9:34.TstCmd.WrtMsg-All entities on layer dim moved to layer door 
9:35.TstCmd.WrtMsg-This sample imports a block from a drawing file 
9:37.TstCmd.WrtMsg-Block sink imported 
9:37.TstCmd.WrtMsg-This sample opens a drawing from file and counting the BlockReferences in the model space 
9:38.TstCmd.WrtMsg-This sample opens a drawing from file and counting the BlockReferences in the model space 
9:38.TstCmd.WrtMsg-Model space block references in file C:\Downloads\blocks_and_tables_-_imperial.dwg: 168 
9:38.TstCmd.WrtMsg-This sample picks an Entity and saves a string on it 
9:38.TstCmd.WrtMsg-Key-value-pair L2A:thi saved on Entity 
9:39.TstCmd.WrtMsg-This sample picks an Entity and reads a string from it 
9:39.TstCmd.WrtMsg-String thi read from Entity 
9:39.TstCmd.WrtMsg-This sample picks an Entity and reads a string from it (with XData as the data source) 
9:40.TstCmd.WrtMsg-This sample counts the number of entities in all paper space layouts 
9:40.TstCmd.WrtMsg-235 entities in all paper space layouts 
9:40.TstCmd.WrtMsg-This sample changes the summary info 
9:40.TstCmd.WrtMsg-Summary info updated 
9:40.TstCmd.WrtMsg-This sample reloads all loaded XRefs

**Reply by wtertinek** [reply]
Thanks a ton for your help! I'll start integrating 2025 ASAP. I'll keep you updated.
